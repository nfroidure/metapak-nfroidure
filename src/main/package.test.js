'use strict';

const packageTransformer = require('./package');

describe('Package transformer', () => {
  it('should work with an empty package.json', () => {
    expect(packageTransformer({})).toMatchSnapshot();
  });

  it('should work with a root package', () => {
    expect(
      packageTransformer({
        metapak: {
          data: {
            rootPackage: true,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should work with a child package.json', () => {
    expect(
      packageTransformer({
        metapak: {
          data: {
            childPackage: true,
          },
        },
      }),
    ).toMatchSnapshot();
  });

  it('should let existing lin & test scripts', () => {
    expect(
      packageTransformer({
        scripts: {
          test: 'echo "Testing is doubting"',
          lint: 'echo "Linting is fearing"',
        },
        metapak: {
          data: {
            childPackage: true,
          },
        },
      }),
    ).toMatchSnapshot();
  });
});
