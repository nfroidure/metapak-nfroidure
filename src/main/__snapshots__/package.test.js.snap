// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Package transformer should let existing lin & test scripts 1`] = `
Object {
  "author": Object {
    "email": "nicolas.froidure@insertafter.com",
    "name": "Nicolas Froidure",
    "url": "https://insertafter.com/en/index.html",
  },
  "contributors": Array [],
  "engines": Object {
    "node": ">=12.19.0",
  },
  "files": Array [
    "src",
    "LICENSE",
    "README.md",
    "CHANGELOG.md",
  ],
  "license": "MIT",
  "metapak": Object {
    "data": Object {
      "childPackage": true,
    },
  },
  "scripts": Object {
    "cli": "env NODE_ENV=\${NODE_ENV:-cli}",
    "lint": "echo \\"Linting is fearing\\"",
    "test": "echo \\"Testing is doubting\\"",
  },
  "version": "0.0.0",
}
`;

exports[`Package transformer should work with a child package.json 1`] = `
Object {
  "author": Object {
    "email": "nicolas.froidure@insertafter.com",
    "name": "Nicolas Froidure",
    "url": "https://insertafter.com/en/index.html",
  },
  "contributors": Array [],
  "engines": Object {
    "node": ">=12.19.0",
  },
  "files": Array [
    "src",
    "LICENSE",
    "README.md",
    "CHANGELOG.md",
  ],
  "license": "MIT",
  "metapak": Object {
    "data": Object {
      "childPackage": true,
    },
  },
  "scripts": Object {
    "cli": "env NODE_ENV=\${NODE_ENV:-cli}",
    "lint": "echo \\"WARNING: No linter specified\\"",
    "test": "echo \\"WARNING: No tests specified\\"",
  },
  "version": "0.0.0",
}
`;

exports[`Package transformer should work with a root package 1`] = `
Object {
  "author": Object {
    "email": "nicolas.froidure@insertafter.com",
    "name": "Nicolas Froidure",
    "url": "https://insertafter.com/en/index.html",
  },
  "config": Object {
    "commitizen": Object {
      "path": "./node_modules/cz-conventional-changelog",
    },
  },
  "contributors": Array [],
  "devDependencies": Object {
    "commitizen": "^4.1.2",
    "conventional-changelog-cli": "^2.1.0",
    "cz-conventional-changelog": "^3.2.0",
  },
  "engines": Object {
    "node": ">=12.19.0",
  },
  "files": Array [
    "src",
    "LICENSE",
    "README.md",
    "CHANGELOG.md",
  ],
  "greenkeeper": Object {
    "ignore": Array [
      "commitizen",
      "cz-conventional-changelog",
      "conventional-changelog-cli",
    ],
  },
  "license": "MIT",
  "metapak": Object {
    "data": Object {
      "rootPackage": true,
    },
  },
  "scripts": Object {
    "changelog": "conventional-changelog -p angular -i CHANGELOG.md -s && git add CHANGELOG.md",
    "cli": "env NODE_ENV=\${NODE_ENV:-cli}",
    "cz": "env NODE_ENV=\${NODE_ENV:-cli} git cz",
    "lint": "lerna run lint",
    "precz": "npm t && npm run lint && npm run metapak -- -s",
    "preversion": "npm t && npm run lint && npm run metapak -- -s",
    "test": "lerna run test",
    "version": "npm run changelog",
  },
  "version": "0.0.0",
}
`;

exports[`Package transformer should work with an empty package.json 1`] = `
Object {
  "author": Object {
    "email": "nicolas.froidure@insertafter.com",
    "name": "Nicolas Froidure",
    "url": "https://insertafter.com/en/index.html",
  },
  "config": Object {
    "commitizen": Object {
      "path": "./node_modules/cz-conventional-changelog",
    },
  },
  "contributors": Array [],
  "devDependencies": Object {
    "commitizen": "^4.1.2",
    "conventional-changelog-cli": "^2.1.0",
    "cz-conventional-changelog": "^3.2.0",
  },
  "engines": Object {
    "node": ">=12.19.0",
  },
  "files": Array [
    "src",
    "LICENSE",
    "README.md",
    "CHANGELOG.md",
  ],
  "greenkeeper": Object {
    "ignore": Array [
      "commitizen",
      "cz-conventional-changelog",
      "conventional-changelog-cli",
    ],
  },
  "license": "MIT",
  "scripts": Object {
    "changelog": "conventional-changelog -p angular -i CHANGELOG.md -s && git add CHANGELOG.md",
    "cli": "env NODE_ENV=\${NODE_ENV:-cli}",
    "cz": "env NODE_ENV=\${NODE_ENV:-cli} git cz",
    "lint": "echo \\"WARNING: No linter specified\\"",
    "precz": "npm t && npm run lint && npm run metapak -- -s",
    "preversion": "npm t && npm run lint && npm run metapak -- -s",
    "test": "echo \\"WARNING: No tests specified\\"",
    "version": "npm run changelog",
  },
  "version": "0.0.0",
}
`;
